airbyte-integrations/connectors/source-klarna/source_klarna/manifest.yaml

```
@@ -0,0 +1,133 @@
+version: "0.29.0"
+
+type: DeclarativeSource
+check:
+  type: CheckStream
+  stream_names:
+    - payouts
+streams:
+  - type: DeclarativeStream
+    name: payouts
+    primary_key:
+      - payout_date
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $schema: http://json-schema.org/draft-07/schema#
+        additionalProperties: true
+        properties: {}
+        type: object
+    retriever:
+      type: SimpleRetriever
+      requester:
+        type: HttpRequester
+        url_base: https://api{{ '-'+config.region if config.region != 'eu' }}.{{ 'playground.' if config.playground }}klarna.com/
+        path: /settlements/v1/payouts
+        http_method: GET
+        request_parameters: {}
+        request_headers: {}
+        authenticator:
+          type: BasicHttpAuthenticator
+          username: '{{ config[''username''] }}'
+          password: '{{ config[''password''] }}'
+        request_body_json: {}
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - payouts
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestPath
+        page_size_option:
+          inject_into: request_parameter
+          type: RequestOption
+          field_name: size
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 500
+          cursor_value: '{{ response.get("pagination", {}).get("next", {}) }}'
+          stop_condition: '{{ not response.get("pagination", {}).get("next", {}) }}'
```

```suggestion
          cursor_value: '{{ response.get("pagination", {}).get("next") }}'
          stop_condition: '{{ not response.get("pagination", {}).get("next") }}'
```

@girarda can you confirm it's better to try to get the item and have None or have an empty `{}` has the same result?

---

either will work since we don't explicitly check for None https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/strategies/cursor_pagination_strategy.py#L55-L60